test/Microsoft.AspNet.Mvc.TagHelpers.Test/EnvironmentTagHelperTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Test
+{
+    public class EnvironmentTagHelperTest
+    {
+        [Theory]
+        [InlineData("Development")]
+        [InlineData("development")]
+        [InlineData("DEVELOPMENT")]
+        [InlineData(" development")]
+        [InlineData("development ")]
+        [InlineData(" development ")]
+        [InlineData("Development,Production")]
+        [InlineData("Production,Development")]
+        [InlineData("Development , Production")]
+        [InlineData(" Development,Production ")]
+        [InlineData("Development,Staging,Production")]
+        [InlineData("Staging,Development,Production")]
+        [InlineData("Staging,Production,Development")]
+        public void ShowsContentWhenCurrentEnvironmentIsSpecified(string namesAttribute)
+        {
+            ShouldShowContent(namesAttribute);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(null)]
+        [InlineData("  ")]
+        [InlineData(", ")]
+        [InlineData("   , ")]
+        [InlineData(",")]
+        [InlineData(",,")]
+        [InlineData(",,,")]
+        [InlineData(",,, ")]
+        public void ShowsContentWhenNoEnvironmentIsSpecified(string namesAttribute)
```

Comment:
There's a trade off when writing unit tests of factoring code like you would in production code, vs. factoring code for fast recognition of the root cause when a test fails. Having a separate test method here with a descriptive name makes it very clear when it fails what the failing case is.